lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -331,47 +336,125 @@ def run_batch(
+                bill_to_user=self._check_is_string(
+                    job_reqs.get(BILL_TO_USER), "bill_to_user"
+                ),
+                ignore_concurrency_limits=bool(job_reqs.get(IGNORE_CONCURRENCY_LIMITS)),
+                scheduler_requirements=job_reqs.get(_SCHEDULER_REQUIREMENTS),
+                debug_mode=norm.get(DEBUG_MODE),
+            )
+        except IncorrectParamsException as e:
+            self._rethrow_incorrect_params_with_error_prefix(e, err_prefix)
+        if perm_type != RequirementsType.STANDARD and not is_write_admin:
+            raise AuthError(
+                f"{err_prefix}In order to specify job requirements you must be a full admin"
+            )
+
+    def _check_is_string(self, putative_str, name):
+        if not putative_str:
+            return None
+        if type(putative_str) != str:
+            raise IncorrectParamsException(f"{name} must be a string")
+        return putative_str
+
+    def _rethrow_incorrect_params_with_error_prefix(
+        self, error: IncorrectParamsException, error_prefix: str
+    ):
+        if not error_prefix:
+            raise error
+        raise IncorrectParamsException(f"{error_prefix}{error.args[0]}") from error
 
     def _check_job_arguments(self, jobs, has_parent_job=False):
         # perform sanity checks before creating any jobs, including the parent job for batch jobs
         for i, job in enumerate(jobs):
             # Could make an argument checker method, or a class that doesn't require a job id.
             # Seems like more code & work for no real benefit though.
             # Just create the class for checks, don't use yet
-            JobSubmissionParameters(
-                "fakejobid",
-                AppInfo(job.get(_METHOD), job.get(_APP_ID)),
-                job[_JOB_REQUIREMENTS],
-                UserCreds(self.sdkmr.get_user_id(), self.sdkmr.get_token()),
-                wsid=job.get(_WORKSPACE_ID),
-                source_ws_objects=job.get(_SOURCE_WS_OBJECTS),
-            )
+            pre = f"Job #{i + 1}: " if len(jobs) > 1 else ""
+            try:
+                JobSubmissionParameters(
+                    "fakejobid",
+                    AppInfo(job.get(_METHOD), job.get(_APP_ID)),
+                    job[_JOB_REQUIREMENTS],
+                    UserCreds(self.sdkmr.get_user_id(), self.sdkmr.get_token()),
+                    wsid=job.get(_WORKSPACE_ID),
+                    source_ws_objects=job.get(_SOURCE_WS_OBJECTS),
+                )
+            except IncorrectParamsException as e:
+                self._rethrow_incorrect_params_with_error_prefix(e, pre)
             if has_parent_job and job.get(_PARENT_JOB_ID):
-                pre = f"Job #{i + 1}: b" if len(jobs) > 1 else "B"
                 raise IncorrectParamsException(
-                    f"{pre}atch jobs may not specify a parent job ID"
+                    f"{pre}batch jobs may not specify a parent job ID"
```

Comment:
Sorry AJ, maintaining proper capitalization was just not worth the effort

Reply:
😱